src/classes/handlers/CaseTriggerHandler.cls

```diff
@@ -0,0 +1,14 @@
+/**
+ * Created by Mariya on 25.01.2023.
+ */
+
+public with sharing class CaseTriggerHandler {
+
+    public static void handler(List<Case> newCases, Map<Id, Case> oldCasesByIds, System.TriggerOperation triggerOperation) {
+        switch on triggerOperation {
+            when BEFORE_UPDATE {
```

Comment:
CRITICAL
I have not noticed this one.
Trigger context is wrong here.
Please study about context and how to properly use them:
https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_triggers_context_variables_considerations.htm